### PR TITLE
Add design document proposal for noop management interface support

### PR DESCRIPTION
**Pull request by rhjanders**
This change adds a design document requested in [PR811](https://github.com/metal3-io/baremetal-operator/pull/811) for further discussion

### CODE REVIEW

File: design/baremetal-operator/noop-management.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+settings that suit my needs.
+
+## Design Details
+
+Add a new `NoopManagement` attribute to the definitions of BMCs which support
+noop management interface. Similar changes will be required in the Custom
+Resource Definition, `baremetalhost_types`, `make-bm-worker` and BMO
+templates so that the request to create the node with `noop`
+`management_interface` is passed to Ironic.
+
```
Comment (hardys) [inline design/baremetal-operator/noop-management.md:80]: This section covers the implementation approach, but it'd be useful to add an example to clearly illustrate the new BMH interface.

**Comment by hardys** [comment]
Overall this LGTM - on the implementation PR one of the concerns was the BMH API (e.g new bmc field vs additional BMC URL types) - so I think it would be useful to add a section/example that describes the new API and behavior, in addition to the implementation details already covered.

File: design/baremetal-operator/noop-management.md (hunk 1 above)
Comment (rhjanders) [reply design/baremetal-operator/noop-management.md:80]: Thank you for your feedback Steve and apologies for the delay in responding. Is this example sufficient, or would you like me to provide further information?

**Pull request closed without merging**